11_dictionary/dictionary.rb

```diff
@@ -0,0 +1,52 @@
+class Dictionary
+  attr_reader :hash
+
+  def initialize
+    @hash = {}
+  end
+
+  def entries
+    @hash
+  end
+
+  def add(key)
+    if key.is_a? String
+      @hash[key] = nil
+    else
+      key.each do |k, v|
+        @hash[k] = v
+      end
+    end
+  end
+
+  def keywords()
+    key_array = []
+    @hash.each { |k, v| key_array.push(k) }
+    key_array.sort
```

Comment:
There's a method in Ruby that allows you to fetch all the keys for a Hash, try to use that.